lib/hyacinth/adapters/preservation_adapter/fedora3/preserved_resource_properties.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module Adapters
+    module PreservationAdapter
+      class Fedora3::PreservedResourceProperties
+        include Fedora3::DatastreamMethods
+        include Fedora3::PropertyContextInitializers
+
+        def to(fedora_obj)
+          return unless @hyacinth_obj.is_a? ::DigitalObject::Asset
+          @hyacinth_obj.resource_attribute_names.each do |resource_name|
+            resource_name = resource_name.to_s
+            next unless @hyacinth_obj.send :"has_#{resource_name}_resource?"
+            resource = @hyacinth_obj.send :"#{resource_name}_resource"
+            resource_config = @hyacinth_obj.class.resource_attributes.fetch(resource_name.to_sym, {})
+            preservable_config = resource_config.fetch(:preservable, {})
+            next unless preservable_config.present? && resource.content_exists?
+            dsid = adapter.dsids_for_resources.fetch(resource_name, resource_name)
+            ensure_datastream(fedora_obj, dsid, datastream_props_for(resource_name, preservable_config, resource))
+            datastream = fedora_obj.datastreams[dsid]
+            # Only update content if it has changed
+            next if "sha256:#{datastream.checksum}" == resource.checksum
```

Comment:
I'm not sure if this is a problem or not, but technically if you change a resource_attribute persist config from :copy to :reference (or vice versa), I think this line will block that update from happening if the resource's checksum hasn't changed.

Reply:
if you change a resource from `:copy` to `:reference` after it has been preserved, you will need to do a migration like the one we did with AC data recently!

Reply:
Hopefully it'll be easier though, since the big difference between that migration and this situation is that Hyacinth will always be the owner of the original copy.  Fedora 3, as a preservation target, is just being told to track Hyacinth's copy for 'E' or make yet another copy for 'M', so I was thinking of it as different.

There may be complications related to Fedora 3 datastream versioning, but I imagined that a switch from :copy to :reference would involve telling Fedora to delete the managed copy and then track Hyacinth's file on the filesystem.  And the opposite would be even easier, since it would be telling Fedora to switch from tracking a file to managing a copy.

The concern in my original comment was just that a Hyacinth developer might change a resource_attribute configuration parameter and not realize that something had gone wrong (or that a manual migration was required) as long as the file checksum hadn't changed.  But we can document that.

Reply:
Another option would be to use the configuration of the datastream (which will match class property if new) to determine content assignment behavior, and then ticket a change to delete and replace if class property has changed.